### PR TITLE
add debug info in multiple-pipeline-playback/capture to root cause #472

### PR DESCRIPTION
**Pull request by aiChaoSONG**

### CODE REVIEW

**Review by marc-hb — CHANGES_REQUESTED** [review]
Please help review #497 (and others) first.

Also let's not add too much to the already ridiculous amount of copy/paste across these two files (they seem 90% identical!!!). You wrote a function so it should be re-usable without copy/paste/diverge...

**Review by marc-hb — CHANGES_REQUESTED** [review]
Sorry I just realized you are reducing copy/paste/diverge a bit: from 4 instances to 2 instances. How about finishing the job and going to just one instance?

**Comment by xiulipan** [comment]
@aiChaoSONG Ping for update, also rebase is needed.

**Comment by aiChaoSONG** [comment]
@xiulipan @marc-hb Updated, I going to merge the multiple-pipeline-playback.sh and multiple-pipeline-capture.sh into one script later, so some common code is still there. the function func_error_exit can be used by other scripts later,so it is moved to lib.sh

File: test-case/multiple-pipeline-playback.sh [hunk 1]
```diff
@@ -92,21 +90,25 @@ func_run_pipeline_with_type()
...
 }
 
 for i in $(seq 1 $loop_cnt)
 do
     dlogi "===== Testing: (Loop: $i/$loop_cnt) ====="
     # clean up dmesg
     sudo dmesg -C
+    # this variable is modified in func_run_pipeline_with_type,
+    # need to reassign at every iteration
+    tmp_count=$max_count
```
Comment (xiulipan) [inline test-case/multiple-pipeline-playback.sh:111]: use local for the tmp_count if you do not want it to be used outside.

File: test-case/multiple-pipeline-playback.sh [hunk 2]
```diff
@@ -92,21 +90,25 @@ func_run_pipeline_with_type()
...
-    pgrep -a arecord
-    pkill -9 aplay
-    pkill -9 arecord
-    exit 1
+    arecord_count=$(pidof arecord | wc -w)
+    dlogi "$arecord_count capture process is running"
+    aplay_count=$(pidof aplay | wc -w)
+    dlogi "$aplay_count playback process is running"
+    overall_count=$((arecord_count + aplay_count))
+    [[ $overall_count -eq $max_count ]] ||
```
Comment (xiulipan) [inline test-case/multiple-pipeline-playback.sh:100]: use `local` for the `overall_count`

**Review by xiulipan — CHANGES_REQUESTED** [review]
try to use local in loops or functions to avoid wrong usage.

**Comment by marc-hb** [comment]
> Please help review #497 (and others) first.

I screwed up #497, apologies. Tentative (and urgent) fix in #521. 

It's not practical to work on the same code at the same time because it will cause git conflicts. @aiChaoSONG please pause your work on the multiple pipelines tests for a couple days until I'm done with:
- urgent fix #521 
- waiting a day or two to make sure I didn't screw up even more
- re-synchronizing multiple-pipeline-playback with multiple-pipeline-capture (I totally missed the massive copy/paste)

Once I'm all done I promise these tests will be all yours to deduplicate and clean-up! Appreciated.

**Comment by marc-hb** [comment]
> re-synchronizing multiple-pipeline-playback with multiple-pipeline-capture (I totally missed the massive copy/paste)

Submitted in #525

**Comment by marc-hb** [comment]
I think I got the root cause, see candidate fix in #538

**Comment by aiChaoSONG** [comment]
@472 is already root caused and fixed, close.

**Pull request closed without merging**